ast/builtins.go

```diff
@@ -452,6 +459,65 @@ var ToNumber = &Builtin{
 	),
 }
 
+// ToArray checks the underlying type of the input. If it is array or set, an array
+// containing the values is returned. If it is not an array, an error is thrown.
+var ToArray = &Builtin{
+	Name: "to_array",
```

Comment:
I would prefer to have these called "cast_<type>" instead of "to_<type>".

As far as I understand, these built-ins are intended to let the policy author cast dynamically typed values (such as the result of an HTTP call out) to a typed value which let's the compiler check them. I think this is a fine solution for simple cases.

Since these built-ins are intended to cast dynamically typed values to more specific types, I don't expect them to perform any kind of fancy conversion like `to_number` does today (or, e.g., coercing a number into a string, like `format_int` does.) I think that `cast_<type>` communicates the intent a bit better.

For what it's worth, in more complex cases, we might want to consider a schema-driven approach that doesn't rely on the policy author converting each individual value to the desired type.